form-flow/src/main/kotlin/com/ritense/formflow/domain/instance/FormFlowInstance.kt

```diff
@@ -142,6 +150,7 @@ class FormFlowInstance(
             this.currentFormFlowStepInstanceId = null
             return null
         }
+        history.removeAll { it.order >= nextStep.order }
```

Comment:
What we discussed about this in the past is that we want to keep the history as long as it's relevant. this means that we want to only remove history for future steps after going back when another "path" is chosen (i.e. because of conditions another nextstep is activated. This prevents data loss when going back and then to next step again.